phoenix-core/src/main/java/org/apache/calcite/jdbc/PhoenixCalciteFactory.java

```diff
@@ -161,6 +162,29 @@ public Object apply(
             }));
         }
 
+        public CalciteStatement createStatement(String sql, int resultSetType, int resultSetConcurrency, int resultSetHoldability) throws SQLException {
+            try {
+                return super.createStatement(resultSetType, resultSetConcurrency, resultSetHoldability);
+            } catch (SQLException e) {
+                if (e.getCause().getCause() instanceof SQLException) {
+                    throw (SQLException) e.getCause().getCause();
```

Comment:
I wondered about this myself. While I didn't encounter any cases where exception levels varied, it's a good idea to account for it. Will make these changes, thanks @maryannxue